cni/pkg/log/uds_test.go

```diff
@@ -56,20 +59,28 @@ func TestUDSLog(t *testing.T) {
 	w.Close()
 	out, _ := io.ReadAll(r)
 
-	// For each level, there should be two lines, one from direct log,
-	// the other one from UDS server
-	wantLevels := []string{"debug", "info", "warn", "error", "debug", "info", "warn", "error"}
-	gotLogs := strings.Split(
-		strings.TrimSuffix(string(out), "\n"), "\n")
-	if want, got := len(wantLevels), len(gotLogs); want != got {
-		t.Fatalf("Number of logs want %v, got %v logs: %v", want, got, gotLogs)
-	}
+	err := retry.UntilSuccess(func() error {
```

Comment:
In L53 after calling Sync(), all logs should be flushed. There is a race, and the fix should be targeted there instead of retrying?

Reply:
+1